php/tests/VerifierTest.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+use Ramsey\Uuid\Uuid;
+use TrueLayer\Signing\Signer;
+use TrueLayer\Signing\Verifier;
+use TrueLayer\Signing\Tests\MockData;
+
+it('should validate a valid signature', function () {
+    $keys = MockData::generateKeyPair();
+    $signer = Signer::signWithKey(Uuid::uuid4(), $keys['private'], null);
+    $verifier = Verifier::verifyWithKey($keys['public']);
+
+    $signature = $signer->method("PUT")
+        ->path("/test")
+        ->header("X-Idempotency-Key", "idempotency-test")
+        ->body('{"random-key": "random-value"}')
+        ->sign();
+
+    $verifier->method('PUT')
+        ->path('/test')
+        ->header('X-Idempotency-Key', 'idempotency-test')
+        ->body('{"random-key": "random-value"}')
+        ->requireHeaders([
+            'X-Idempotency-Key'
+        ]);
+
+    expect($verifier->verify($signature))->not->toThrow(Exception::class);
+});
+
+it('should throw when required header is missing', function () {
```

Comment:
Could do the other verify scenario like "method mismatch" etc, but not a huge deal.

Reply:
Done!